### PR TITLE
Fix `Discount.getFreeBurger` exercise

### PR DESCRIPTION
**Pull request by feihong**
- [x] Move the 'variant constructor is not a function' exercise into the main text
- [x] Fix the overly-complicated solution for exercise 1
- [x] Add another exercise to simplify `Discount.getFreeBurger` into its final form

### CODE REVIEW

File: docs/burger-discounts/Discount.re [hunk 1]
```diff
@@ -160,13 +160,34 @@ let getFreeBurger = (items: array(Item.t)) => {
...
-  | (Some(_), Some(cheaperPrice)) => Some(cheaperPrice)
-  | (None | Some(_), None | Some(_)) => None
+  switch (prices[1]) {
+  | None => None
+  | Some(cheaperPrice) => Some(cheaperPrice)
   };
 };
 // #endregion improved-get-free-burger
 
+ignore(getFreeBurger);
```
Comment (jchavarri) [inline docs/burger-discounts/Discount.re:170]: Why is this necessary? To remove some "unused" warning I assume?

**Review by jchavarri — APPROVED** [review]

File: docs/burger-discounts/Discount.re (hunk 1 above)
Comment (feihong) [reply docs/burger-discounts/Discount.re:170]: Yeah, I have to use `ignore` quite a bit in source files whose code regions are imported into the text
